test/integration/replicationcontroller/replicationcontroller_test.go

```diff
@@ -154,6 +154,22 @@ func rmSetup(t *testing.T, enableGarbageCollector bool) (*httptest.Server, *repl
 	return s, rm, podInformer, clientSet
 }
 
+// wait for the podInformer to observe the pods. Call this function before
+// running the RC manager to prevent the rc manager from creating new pods
+// rather than adopting the existing ones.
+func waitToObservePods(t *testing.T, podInformer controllerframwork.SharedIndexInformer, podNum int) {
```

Comment:
Won't block the fix on this, but it would be good to move this somewhere common so it's not duplicated.